File: src/main/java/com/wlyscraft/slimefunitemexpansion/machines/LavaFactory.java
package com.wlyscraft.slimefunitemexpansion.machines;

import io.github.thebusybiscuit.slimefun4.core.attributes.RecipeDisplayItem;
import io.github.thebusybiscuit.slimefun4.implementation.SlimefunItems;
import me.mrCookieSlime.Slimefun.Lists.RecipeType;
import me.mrCookieSlime.Slimefun.Objects.Category;
import me.mrCookieSlime.Slimefun.Objects.SlimefunItem.abstractItems.AContainer;
import me.mrCookieSlime.Slimefun.api.SlimefunItemStack;
import org.bukkit.Material;
import org.bukkit.inventory.ItemStack;

import javax.annotation.Nonnull;

public class LavaFactory extends AContainer implements RecipeDisplayItem {
    public LavaFactory(Category category, SlimefunItemStack item, RecipeType recipeType, ItemStack[] recipe) {
        super(category, item, recipeType, recipe);
    }

    @Override
    public ItemStack getProgressBar() {
        return SlimefunItems.LAVA_CRYSTAL;
    }

    @Nonnull
    @Override
    public String getMachineIdentifier() {
        return "IE_LAVA_FACTORY";
    }

    @Override
    public void registerDefaultRecipes() {
        this.registerRecipe(25, new ItemStack[]{new ItemStack(Material.BUCKET)}, new ItemStack[]{new ItemStack(Material.LAVA_BUCKET)});
    }
}

File: src/main/kotlin/io/github/itsflicker/slimefunitemexpansion/items/machines/LavaFactory.kt
package io.github.itsflicker.slimefunitemexpansion.items.machines

import io.github.thebusybiscuit.slimefun4.api.items.ItemGroup
import io.github.thebusybiscuit.slimefun4.api.items.SlimefunItemStack
import io.github.thebusybiscuit.slimefun4.api.recipes.RecipeType
import io.github.thebusybiscuit.slimefun4.core.attributes.RecipeDisplayItem
import io.github.thebusybiscuit.slimefun4.implementation.SlimefunItems
import me.mrCookieSlime.Slimefun.Objects.SlimefunItem.abstractItems.AContainer
import org.bukkit.Material
import org.bukkit.inventory.ItemStack

class LavaFactory(itemGroup: ItemGroup, item: SlimefunItemStack, recipeType: RecipeType, recipe: Array<ItemStack?>) :
    AContainer(itemGroup, item, recipeType, recipe), RecipeDisplayItem {

    override fun getProgressBar(): ItemStack {
        return SlimefunItems.LAVA_CRYSTAL
    }

    override fun getMachineIdentifier(): String {
        return "IE_LAVA_FACTORY"
    }

    public override fun registerDefaultRecipes() {
        this.registerRecipe(30, arrayOf(ItemStack(Material.BUCKET)), arrayOf(ItemStack(Material.LAVA_BUCKET)))
    }
}